Sources/Workspace/InitPackage.swift

```diff
@@ -350,8 +381,79 @@ public final class InitPackage {
         }
     }
 
+    private func writePlugins() throws {
+        switch packageType {
+        case .buildToolPlugin, .commandPlugin:
+            let plugins = destinationPath.appending(component: "Plugins")
+            guard self.fileSystem.exists(plugins) == false else {
+                return
+            }
+            progressReporter?("Creating \(plugins.relative(to: destinationPath))/")
+            try makeDirectories(plugins)
+
+            let moduleDir = plugins
+            try makeDirectories(moduleDir)
+
+            let sourceFileName = "\(pkgname).swift"
+            let sourceFile = try AbsolutePath(validating: sourceFileName, relativeTo: moduleDir)
+
+            var content = """
+                import PackagePlugin
+
+                @main
+
+                """
+            if packageType == .buildToolPlugin {
+                content += """
+                struct \(typeName): BuildToolPlugin {
+                    func createBuildCommands(context: PluginContext, target: Target) async throws -> [Command] {
+                        // The plugin can choose what parts of the package to process.
+                        guard let target = target as? SourceModuleTarget else { return [] }
```

Comment:
I think we have a nice API for this? @neonichu ?

Reply:
We do have 

```
public extension Target {
    /// Convenience accessor which casts the receiver to`SourceModuleTarget` if possible.
    var sourceModule: SourceModuleTarget? {
        return self as? SourceModuleTarget
    }
}
```

Reply:
So we do this instead?
```
    let sourceFiles = target.sourceModule?.sourceFiles ?? []
```

And then proceed as if the target had source files even if it doesn't?

Reply:
Actually that doesn't work since one is a `FileList?` and the other an array.

So just `let sourceFiles = target.sourceModule?.sourceFiles` I guess and then an optional unwrap in the iterator.